osu.Server.Queues.ScoreStatisticsProcessor/Processors/UserTotalPerformanceProcessor.cs

```diff
@@ -21,12 +21,26 @@ public class UserTotalPerformanceProcessor : IProcessor
         private BeatmapStore? beatmapStore;
         private BuildStore? buildStore;
 
+        // This processor needs to run after the score's PP value has been processed.
+        public int Order => ScorePerformanceProcessor.ORDER + 1;
+
         public void RevertFromUserStats(SoloScoreInfo score, UserStats userStats, int previousVersion, MySqlConnection conn, MySqlTransaction transaction)
         {
         }
 
         public void ApplyToUserStats(SoloScoreInfo score, UserStats userStats, MySqlConnection conn, MySqlTransaction transaction)
         {
+            var dbInfo = LegacyDatabaseHelper.GetRulesetSpecifics(score.ruleset_id);
+
+            UpdateUserStatsAsync(userStats, score.ruleset_id, conn, transaction).Wait();
+
+            int warnings = conn.QuerySingleOrDefault<int>($"SELECT `user_warnings` FROM {dbInfo.UsersTable} WHERE `user_id` = @UserId", new
```

Comment:
I don't understand this. Shouldn't it just skip the `UpdateUserStatsAsync` call rather than zero the value out? Is there a reason to firs calculate and then reset?

Reply:
Nope, I had it written this way in another branch and didn't double-take.